src/core/buffer.cc

```diff
@@ -176,6 +178,20 @@ class BufferImpl
       return contains_pyobjects_;
     }
 
+    bool is_page_multiple() const noexcept {
+      return !(size_%page_size_);
+    }
+
+    static size_t calc_page_size() {
+      #if DT_OS_WINDOWS
+        SYSTEM_INFO sysInfo;
+        GetSystemInfo(&sysInfo);
+        const size_t page_size = sysInfo.dwPageSize;
+      #else
+        const size_t page_size = static_cast<size_t>(sysconf(_SC_PAGE_SIZE));
```

Comment:
no need for a local variable - can return the result immediately

Reply:
That's true. But I normally prefer to have just one `return` statement, in this way it's easier to control what and where we actually return.

Reply:
Yeah, but given that this is a `#if .. #else` case, the compiler will really see only 1 return anyways.

Reply:
For compiler, yes. I'm talking more about myself looking at the code :) It seems easier to catch just one `return` line than going through all the macro branches.